lib/model.js

```diff
@@ -1406,7 +1406,7 @@ class Model {
       override: false
     }, options);
 
-    if ((name === 'defaultScope' || name in this.options.scopes) && options.override === false) {
+    if ((name === 'defaultScope'  && Object.keys(this.options.defaultScope).length > 0 || name in this.options.scopes) && options.override === false) {
```

Comment:
Thank you for your help and advice. I'm a little confused. The code you have written in response would throw a parsing error because of the extra paren. Also, adding another pair of parentheses like `(((name === 'defaultScope' && Object.keys(this.options.defaultScope).length > 0) || name in this.options.scopes) && options.override === false)` seems unnecessary. I think `((name === 'defaultScope' && Object.keys(this.options.defaultScope).length > 0 || name in this.options.scopes) && options.override === false)` works. It passes linting and my tests. Let me know if you think this is still wrong. Thanks :)

Reply:
Yes I mistakenly added one bracket in last as I was typing on Github.

What I want is to avoid checking `defaultScope` keys when `scopeName` is not `defaultScope`, what we currently have will always execute that code for all scopes

Reply:
Hi @sushantdhiman. Not sure if this was closed because it was stale or unneeded, but I was looking back at this PR after looking at the open issue. 

I believe what @aw-davidson is  saying is correct. 

Perhaps I am mistaken, but with lazy evaluation won't the `Object.keys(this.options.defaultScope)` never be evaluated if `name !== 'defaultScope'`? The extra bracket suggested won't change the order of evaluation.

Reply:
@Lawton I just took a fresh look and you and @aw-davidson originally was correct, sorry for overlooking this. There is a linting issue I will fix that an merge this :)

Reply:
Looks like I don't have edit permission on fork, @Lawton if you want you can open a new PR.

You will notice a extra space after `name === 'defaultScope'  `, please get rid of that in new PR and I will merge